website/client/components/shops/buyModal.vue

```diff
@@ -402,6 +407,25 @@
           return;
         }
 
+        if (this.item.pinType === 'premiumHatchingPotion' || this.item.pinType === 'eggs' && dropEggKeys.indexOf(this.item.key) === -1) {
```

Comment:
is this.item.buy false here?

Reply:
Yes--if `this.item.buy` were true, we'd have returned out on line 407.